tiago_hhba_cfg/tiago_perceptions/scripts/proc_keyword_detection.py

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env python
+import os
+import rospy
+import traceback
+import KeywordRecognizer as kr
+from std_msgs.msg import Bool
+import HomoDeUS_common_py.HomoDeUS_common_py as common
+
+class Keyword_detection:
+    """
+    This class publishes the state of detection of the wanted keyword
+    """
+    def __init__(self, keyword='legacy', timeout=30):
+        """
+        This method initializes the perception module by initializing the output topic and
+        the keywordRecognizer object 
+        Arguments
+        ---------
+        keyword : string
+            The keyword used to activate the voice interface of the robot. By
+            default, its value is set to 'legacy', but it can be changed for
+            a word in Pocket Sphinx's dictionary.
+        timeout : int
+            The maximum duration (in seconds) of the wait when the robot is
+            waiting for a keyword to be said. By default, its value is set to
+            30 seconds.
+        """ 
+        #The output of the module
+        self.output_perc = rospy.Publisher("/proc_output_keywordDetect", Bool, queue_size=10)
+
+        self.keyword_recognizer = kr.KeywordRecognizer(keyword=keyword, timeout=timeout)
+
+    def transform(self):
+        while not rospy.is_shutdown():
+            detection = self.keyword_recognizer.wait_for_keyword()
```

Comment:
Ajout d'un commentaire TODO qui lie le wait for keyword à un call back et empêche ainsi lutilisation de CPU pour rien.